internal/controller/suite_test.go

```diff
@@ -127,18 +127,16 @@ var _ = BeforeSuite(func() {
 	k8sClient = k8sManager.GetClient()
 	Expect(k8sClient).ToNot(BeNil())
 
-	err = (&PipelineRolloutReconciler{
-		client:     k8sManager.GetClient(),
-		scheme:     k8sManager.GetScheme(),
-		restConfig: cfg,
-	}).SetupWithManager(k8sManager)
+	err = NewPipelineRolloutReconciler(
+		k8sManager.GetClient(),
+		k8sManager.GetScheme(),
+		cfg).SetupWithManager(k8sManager)
```

Comment:
I preferred not to call `NewPipelineRolloutReconciler` to be able to create a more custom reconciler for testing (ex: with less workers, etc.). However, I am not opposed to this since it reduces the amount of testing code we have to write. It's a bit of a tradeoff. What are you reasons behind this choice?

Reply:
`NewPipelineRolloutReconciler()` starts up the workers to do the reconciliations

Reply:
Plus, it allows testing of this function as well.